scripts/skills/perks/perk_rf_kingfisher.nut

```diff
@@ -35,82 +36,92 @@ this.perk_rf_kingfisher <- ::inherit("scripts/skills/skill", {
 
 	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
 	{
-		if (!_targetEntity.isAlive())
+		if (!_targetEntity.isAlive() || ::Math.rand(1, 100) > this.m.Chance || !this.getContainer().getActor().isPlacedOnMap() || !_targetEntity.isPlacedOnMap())
```

Comment:
You call `this.getContainer().getActor()` several times in this function.
You should therefor use a `local actor` for it.